snapcraft/parts/setup_assets.py

```diff
@@ -77,6 +86,10 @@ def setup_assets(
         relative_icon_path = str(icon_path)
 
     for app_name, app in project.apps.items():
+        _validate_command_chain(
+            app.command_chain, name=f"app {app_name!r}", prime_dir=prime_dir
+        )
+
         if not app.desktop:
             continue
```

Comment:
this continue in the middle has proved probalematic, I would change this now and do `if app.desktop` and nest the desktop file logic below